go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package verticalsplit
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"vitess.io/vitess/go/json2"
+	"vitess.io/vitess/go/vt/vtgate/vtgateconn"
+
+	"github.com/stretchr/testify/assert"
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/test/endtoend/sharding"
+	"vitess.io/vitess/go/vt/proto/topodata"
+	_ "vitess.io/vitess/go/vt/vtgate/grpcvtgateconn"
+)
+
+var (
+	clusterInstance      *cluster.LocalProcessCluster
+	sourceKeyspace       = "source_keyspace"
+	destinationKeyspace  = "destination_keyspace"
+	hostname             = "localhost"
+	cellj                = "test_nj"
+	shardName            = "0"
+	createTabletTemplate = `
+		create table %s(
+			id bigint not null,
+			msg varchar(64),
+			primary key (id),
+			index by_msg (msg)
+		) Engine=InnoDB;`
+	createViewTemplate     = "create view %s(id, msg) as select id, msg from %s;"
+	createMoving3NoPkTable = `
+		create table moving3_no_pk (
+			id bigint not null,
+			msg varchar(64)
+		) Engine=InnoDB;`
+	tableArr         = []string{"moving1", "moving2", "staying1", "staying2"}
+	insertIndex      = 0
+	moving1First     int
+	moving2First     int
+	staying1First    int
+	staying2First    int
+	moving3NoPkFirst int
+)
+
+func TestVerticalSplit(t *testing.T) {
+	// setting RBR mode
+	os.Setenv("EXTRA_MY_CNF", path.Join(os.Getenv("VTROOT"), "config", "mycnf", "rbr.cnf"))
+	// setting RBR mode
+
+	flag.Parse()
+	code, err := initializeCluster(t)
+	if err != nil {
+		t.Errorf("setup failed with status code %d", code)
+	}
+	defer teardownCluster(t)
+
+	// Adding another cell in the same cluster
+	err = clusterInstance.TopoProcess.ManageTopoDir("mkdir", "/vitess/"+"test_ca")
+	err = clusterInstance.VtctlProcess.AddCellInfo("test_ca")
+	err = clusterInstance.TopoProcess.ManageTopoDir("mkdir", "/vitess/"+"test_ny")
+	err = clusterInstance.VtctlProcess.AddCellInfo("test_ny")
```

Comment:
should do assert.Nil(t.err) for all the above statements

Reply:
Done. Refer PR46